src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php

```diff
@@ -184,18 +187,17 @@ public function execute()
      */
     protected function findIdsByGivenCriteria()
     {
-        $select = null;
-        if (null !== $this->idField) {
-            $select = $this->idField->getSelect();
-        }
+        $select = $this->idField->getSelect() . ' AS ' . $this->idField->getName();
 
         $subQueryBuilder = $this->createSubQueryBuilder($select);
         if ($this->limit != null) {
             $subQueryBuilder->setMaxResults($this->limit)->setFirstResult($this->limit * ($this->page - 1));
         }
 
         foreach ($this->sortFields as $index => $sortField) {
-            $subQueryBuilder->addSelect($sortField->getSelect() . ' AS ' . $sortField->getName());
+            if ($sortField != $this->idField) {
```

Comment:
will also change it to check only for the `name` else it could error when using the default idField.

```php
if ($sortField->getName() !== $this->idField->getName()) {
```